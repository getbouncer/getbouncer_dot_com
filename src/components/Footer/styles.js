import styled from '@emotion/styled'
import { Link } from 'gatsby'
import { breakpoints } from '~styles'

export const FooterElem = styled.footer`
  background-color: #fff;
`

export const InnerContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: 30px 0;
  position: relative;

  @media ${breakpoints.smartphone} {
    padding: 45px 0;
  }
`

export const StyledLink = styled(Link)`
  left: 50%;
  position: absolute;
  transform: translateX(-50%);
`

export const NavSocials = styled.ul`
  align-items: center;
  display: flex;
  margin-left: auto;

  li:not(:last-child) {
    margin-right: 14px;
  }

  img {
    height: 26px;
    width: 26px;
  }
`
