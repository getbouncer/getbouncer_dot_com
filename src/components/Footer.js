import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/Footer.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faFacebook } from '@fortawesome/free-brands-svg-icons'
import { faTwitter } from '@fortawesome/free-brands-svg-icons'
import { faSlack } from '@fortawesome/free-brands-svg-icons'

/**
 * Need column before I can continue with this one
 * Will define props which will be separated into columns
 */
class Footer extends React.Component {
  render() {
    return (
      <div className={styles.footer}>
        <div className={styles.links}>
          Terms &amp; Conditions
        </div>
        <div>
          <FontAwesomeIcon icon={faFacebook} size='2x' className={styles.icon} style></FontAwesomeIcon>
          <FontAwesomeIcon icon={faTwitter} size='2x' className={styles.icon}></FontAwesomeIcon>
          <FontAwesomeIcon icon={faSlack} size='2x' className={styles.icon}></FontAwesomeIcon>
        </div>
      </div>
    )
  }
}

Footer.propTypes = {
  fbLink: PropTypes.string,
  twLink: PropTypes.string,
  copyright: PropTypes.string,
  policy: PropTypes.string
};

export default Footer;