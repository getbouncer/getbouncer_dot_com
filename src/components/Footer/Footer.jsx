import React from 'react'
import PropTypes from 'prop-types'
import { useStaticQuery, graphql } from 'gatsby'
import Container from '~globals/Container'
import { AboveSmartphone } from '~components/Media'
import { FooterElem, InnerContainer, StyledLink, NavSocials } from './styles'

export const PureFooter = ({ data }) => {
  const { minimal_logo: minimalLogo, socials } = data

  return (
    <FooterElem>
      <Container isLarge>
        <InnerContainer>
          <StyledLink to="/">
            <img src={minimalLogo.url} alt={minimalLogo.alt} />
          </StyledLink>
          <AboveSmartphone>
            <NavSocials>
              {socials.map(social => (
                <li key={social.social_logo.alt}>
                  <a href={social.social_link.url} target="_blank" rel="noopener noreferrer">
                    <img src={social.social_logo.url} alt={social.social_logo.alt} />
                  </a>
                </li>
              ))}
            </NavSocials>
          </AboveSmartphone>
        </InnerContainer>
      </Container>
    </FooterElem>
  )
}

PureFooter.propTypes = {
  data: PropTypes.shape().isRequired,
}

export const Footer = () => {
  const data = useStaticQuery(graphql`
    query FooterQuery {
      prismicGlobals {
        data {
          minimal_logo {
            alt
            url
          }
          socials {
            social_link {
              url
            }
            social_logo {
              alt
              url
            }
          }
        }
      }
    }
  `)

  return <PureFooter data={data.prismicGlobals.data} />
}

export default Footer
