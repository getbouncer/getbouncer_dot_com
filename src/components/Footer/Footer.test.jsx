import React from 'react'
import renderer from 'react-test-renderer'
import { PureFooter as Footer } from './Footer'

/* eslint-disable no-undef */

describe('Footer', () => {
  it('renders correctly', () => {
    const data = {
      minimal_logo: {
        alt: 'logo',
        url: 'https://www.example.com/logo',
      },
      socials: [
        {
          social_link: {
            url: 'https://www.example.com/social_1',
          },
          social_logo: {
            alt: 'social 1 logo',
            url: 'https://www.example.com/logo_1.svg',
          },
        },
        {
          social_link: {
            url: 'https://www.example.com/social_2',
          },
          social_logo: {
            alt: 'social 2 logo',
            url: 'https://www.example.com/logo_2.svg',
          },
        },
      ],
    }
    const tree = renderer
      .create(<Footer data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
