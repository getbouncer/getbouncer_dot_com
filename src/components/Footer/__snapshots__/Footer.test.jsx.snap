// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="css-709yn1"
>
  <div
    className="css-sdcyqc"
  >
    <div
      className="css-ih8zfl"
    >
      <a
        className="css-1uxekxc"
        href="/"
      >
        <img
          alt="logo"
          src="https://www.example.com/logo"
        />
      </a>
    </div>
  </div>
</footer>
`;
